source/game/field/CourseColMgr.cc

```diff
@@ -0,0 +1,146 @@
+    if (!gndcol) {
+        gndcol = m_gndCol;
+    }
+
+    m_kclScale = scalar;
+    EGG::Vector3f scaled_position = v0 / scalar;
+    EGG::Vector3f vStack88 = v1 / scalar;
+    gndcol->lookupSphere(radius, scaled_position, vStack88, flags);
+
+    if (info) {
+        return doCheckWithFullInfoPush(gndcol, &RKGndCol::checkSphere, info, kcl_flags_out);
+    }
+    return false; // doCheckMaskOnlyPush(gndcol, &RKGndCol::checkSphereCollision,
+                  // kcl_flags_out);
+}
+
+void CourseColMgr::pushCollisionEntry(float dist, KCLTypeMask *typeMask, KCLTypeMask typeBit,
+        u16 attribute) {
+    *typeMask |= typeBit;
+
+    size_t arrSize = CollisionDirector::s_entries.size();
+    if (CollisionDirector::s_collisionEntryCount >= arrSize) {
+        CollisionDirector::s_collisionEntryCount = arrSize - 1;
+    }
+
+    auto &entry = CollisionDirector::s_entries[CollisionDirector::s_collisionEntryCount++];
+    entry.typeMask = typeBit;
+    entry.attribute = attribute;
+    entry.dist = dist;
+}
+
+bool CourseColMgr::doCheckWithFullInfoPush(RKGndCol *colMgr, CollisionCheckFunc collisionCheckFunc,
+        CollisionInfo *colInfo, u32 *flagsOut) {
+    f32 dist;
+    EGG::Vector3f fnrm;
+    u16 attribute;
+    bool hasCol = false;
+
+    while ((colMgr->*collisionCheckFunc)(&dist, &fnrm, &attribute)) {
+        dist *= m_kclScale;
+
+        /*if (m_noBounceWallInfo && (attribute & KCL_SOFT_WALL_MASK)) {
+            if (m_localMtx) {
+                nw4r::math::VEC3TransformNormal(&fnrm, this->localMtx, &fnrm);
+            }
+
+            offset = fnrm * dist;
+            this->softWallColInfo->update(dist, offset, fnrm);
+
+            hasCol = true;
+        } else {*/
+        u32 kclAttributeTypeBit = KCL_ATTRIBUTE_TYPE_BIT(attribute);
+        if (flagsOut) {
+            pushCollisionEntry(dist, flagsOut, kclAttributeTypeBit, attribute);
+        }
+        if (kclAttributeTypeBit & KCL_TYPE_SOLID_SURFACE) {
+            colInfo->update(dist, fnrm * dist, fnrm, kclAttributeTypeBit);
+        }
+        hasCol = true;
+        /*}*/
+    }
+    this->m_localMtx = nullptr;
+    return hasCol;
+}
+
+void CourseColMgr::CollisionInfo::updateFloor(f32 dist, const EGG::Vector3f &fnrm) {
+    if (dist > m_floorDist) {
+        m_floorDist = dist;
+        m_floorNrm = fnrm;
+    }
+}
+
+void CourseColMgr::CollisionInfo::updateWall(f32 dist, const EGG::Vector3f &fnrm) {
+    if (dist > m_wallDist) {
+        m_wallDist = dist;
+        m_wallNrm = fnrm;
+    }
+}
+
+void CourseColMgr::CollisionInfo::update(f32 now_dist, const EGG::Vector3f &offset,
+        const EGG::Vector3f &fnrm, u32 kclAttributeTypeBit) {
+    m_bbox.mMin = m_bbox.mMin.minimize(offset);
+    m_bbox.mMax = m_bbox.mMax.maximize(offset);
+
+    if (kclAttributeTypeBit & KCL_TYPE_FLOOR) {
+        updateFloor(now_dist, fnrm);
+    } else if (kclAttributeTypeBit & KCL_TYPE_WALL) {
+        if (EGG::Mathf::isNonNegative(m_wallDist)) {
```

Comment:
We can remove the change in `egg/math/Math.hh` entirely.
```suggestion
        if (m_wallDist > -std::numeric_limit<f32>::min) {
```

Reply:
You're saying you want to get rid of `EGG::Mathf::isNonNegative`? I feel like that will be used later on. Is it worth keeping from a readability standpoint?